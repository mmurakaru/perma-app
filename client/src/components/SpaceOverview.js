import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { ReactComponent as ArrowIcon } from '../assets/arrow_down.svg'
import { ReactComponent as Plus } from '../assets/plus.svg'
import { ReactComponent as Sunflower } from '../assets/sunflower.svg'

SpaceOverview.propTypes = {
    switchToPlantFields: PropTypes.func,
    spaces: PropTypes.array,
    updateSpaceDetails: PropTypes.func,
    switchToSpaceCreator: PropTypes.func,
}

export default function SpaceOverview({
    switchToPlantFields,
    spaces,
    updateSpaceDetails,
    switchToSpaceCreator,
}) {
    return (
        <BackgroundStyled>
            <SpaceCreatorContainer>
                <ArrowDown
                    data-testid="ArrowDown"
                    onClick={switchToPlantFields}
                />
                <h2>Choose a space</h2>
                <>
                    <SpacesContainer>
                        {spaces.map((title, index) => (
                            <SpaceWrapper
                                key={index}
                                onClick={() => updateSpaceDetails(index)}
                            >
                                <SpaceIcon>
                                    <LogoStyled />
                                </SpaceIcon>
                                <h2>{title}</h2>
                            </SpaceWrapper>
                        ))}

                        <SpaceWrapper onClick={switchToSpaceCreator}>
                            <NewSpaceIcon>
                                <PlusIcon />
                            </NewSpaceIcon>
                            <h2>Create a new space</h2>
                        </SpaceWrapper>
                    </SpacesContainer>
                </>
            </SpaceCreatorContainer>
        </BackgroundStyled>
    )
}

const BackgroundStyled = styled.div`
    background: var(--medium-grey);
    width: 100%;
    height: 100%;
    padding: 20px;
`

const SpaceCreatorContainer = styled.section`
    border-radius: 6px;
    box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
    padding: 20px;
    background-color: var(--white);
    width: 100%;
    overflow: none;
    margin-top: 50px;

    h2 {
        margin-left: 20px;
        font-weight: 400;
        font-size: 0.6rem;
        opacity: 50%;
    }
`
const ArrowDown = styled(ArrowIcon)`
    fill: var(--dark-green);
    width: 20px;
    position: absolute;
    left: 20px;
    top: 20px;
`

const SpacesContainer = styled.div`
    margin-top: 20px;
    height: 65vh;
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-auto-rows: min-content;
    place-items: center;
    gap: 20px;
    overflow: auto;

    .link {
        color: var(--dark-grey);
        text-decoration: none;
    }
`

const NewSpaceIcon = styled.div`
    border: 1px solid var(--medium-grey);
    height: 100px;
    width: 100px;
    border-radius: 5px;
    display: grid;
    place-items: center;
`

const PlusIcon = styled(Plus)`
    width: 50%;
    display: block;
    fill: var(--dark-grey);
`

const SpaceIcon = styled(NewSpaceIcon)`
    border: none;
    background-color: var(--light-green);
`
const LogoStyled = styled(Sunflower)`
    width: 80%;
`

const SpaceWrapper = styled.div`
    display: flex;
    flex-direction: column;

    h2 {
        margin: 3px 0;
        opacity: 100%;
    }
`
