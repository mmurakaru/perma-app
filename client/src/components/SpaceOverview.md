SpaceOverview example:

```js
<SpaceOverview spaceTitles={['Vegetables', 'Fruits']} />
```
