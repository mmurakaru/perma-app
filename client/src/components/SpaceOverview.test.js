import SpaceOverview from './SpaceOverview'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('SpaceOverview', () => {
    it('renders correctly', () => {
        const { container } = render(
            <SpaceOverview spaceTitles={['Vegetables', 'Fruits']} />
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    const onClickMock = jest.fn()

    it('calls onClick', () => {
        const { getByTestId } = render(
            <SpaceOverview
                switchToPlantFields={onClickMock}
                spaceTitles={['Vegetables', 'Fruits']}
            />
        )
        userEvent.click(getByTestId('ArrowDown'), 'Coconut')
        expect(onClickMock).toHaveBeenCalled()
    })
})
