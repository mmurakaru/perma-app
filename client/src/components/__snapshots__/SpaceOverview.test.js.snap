// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceOverview renders correctly 1`] = `
.c0 {
  background: var(--mediumGrey);
  width: 100%;
  height: 100%;
  padding: 20px;
}

.c1 {
  border-radius: 6px;
  box-shadow: rgba(0,0,0,0.24) 0px 3px 8px;
  padding: 20px;
  background-color: var(--white);
  width: 100%;
  overflow: none;
  margin-top: 50px;
}

.c1 h2 {
  margin-left: 20px;
  font-weight: 400;
  font-size: 0.6rem;
  opacity: 50%;
}

.c2 {
  fill: var(--darkGreen);
  width: 20px;
  position: absolute;
  left: 20px;
  top: 20px;
}

.c3 {
  margin-top: 20px;
  height: 65vh;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-auto-rows: min-content;
  place-items: center;
  gap: 20px;
  overflow: auto;
}

.c3 .link {
  color: var(--darkGrey);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  border: 1px solid var(--mediumGrey);
  height: 100px;
  width: 100px;
  border-radius: 5px;
  display: grid;
  place-items: center;
}

.c8 {
  width: 50%;
  display: block;
  fill: var(--darkGrey);
}

.c6 {
  border: none;
  background-color: var(--lightGreen);
}

.c7 {
  width: 80%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 h2 {
  margin: 3px 0;
  opacity: 100%;
}

<div
  class="c0"
>
  <section
    class="c1"
  >
    <svg
      class="c2"
      data-testid="ArrowDown"
    >
      arrow_down.svg
    </svg>
    <h2>
      Choose a space
    </h2>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5 c6"
        >
          <svg
            class="c7"
          >
            banana.svg
          </svg>
        </div>
        <h2>
          Vegetables
        </h2>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5 c6"
        >
          <svg
            class="c7"
          >
            banana.svg
          </svg>
        </div>
        <h2>
          Fruits
        </h2>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <svg
            class="c8"
          >
            plus.svg
          </svg>
        </div>
        <h2>
          Create a new space
        </h2>
      </div>
    </div>
  </section>
</div>
`;
